### ISSUE TITLE
Error: Type 'E[keyof E]' is not assignable to type 'string | number'

### ISSUE THREAD

**Issue by thiagodp** [AnyhowStep/enum-util#3]
I'm receiving the following error from `tsc`:

```bash
node_modules/enum-util/dist/src/main/util.d.ts(6,96): error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'E[keyof E]' is not assignable to type 'string | number'.
    Type 'E[keyof E]' is not assignable to type 'number'.
node_modules/enum-util/dist/src/main/util.d.ts(17,47): error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'E[keyof E]' is not assignable to type 'string | number'.
    Type 'E[keyof E]' is not assignable to type 'number'.
```

**Comment by AnyhowStep** [comment]
What version of typescript are you running?

**Comment by thiagodp** [comment]
`2.7.1`

**Comment by AnyhowStep** [comment]
How did you get this error?

Is there a particular code snippet you're trying compile that gives you this problem?

**Comment by AnyhowStep** [comment]
I just tested and it seems like typescript `2.7` has some new rules about type predicates, which explains why it complains the way you reported. I resigned to making the parameter `any` but the return type will still be `E[keyof E]`, which should be `string` or `number`, depending on the enum

**Comment by AnyhowStep** [comment]
Version `1.2.0` should get rid of these annoying problems.

**Comment by thiagodp** [comment]
Yes, solved in `1.2.0`. Thanks!
